### PR TITLE
Add syntax highlighting support for 11 languages

### PR DESCRIPTION
**Pull request by TanvirOnGH**
Added syntax highlighting support for the following languages:

1. CoffeeScript
2. D
3. Groovy
4. Julia
5. Matlab
6. Nix
7. NuShell
8. Ocaml
9. Processing
10. Raku
11. Zig

### CODE REVIEW

File: syntax.d/coffeescript.ini [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name=CoffeeScript
+extensions=coffee, litcoffee
+highlight_numbers=true
+singleline_comment_start=#
+multiline_comment_delims=###, ###
+singleline_string_quotes="
+multiline_string_delim="
+keywords_1=this, new, if, else, return, try, catch, throw, function, class, extends, super, switch, case, default, break, continue, for, in, while, do, unless, until
+keywords_2=true, false, null, undefined
```
Comment (ilai-deutel) [inline syntax.d/coffeescript.ini:9]: Is there a reference page that lists all keywords for CoffeeScript (and the other languages as well)? If so, can you add it as a comment, like in https://github.com/ilai-deutel/kibi/blob/master/syntax.d/python.ini#L7?

File: syntax.d/groovy.ini [hunk 2]
```diff
@@ -0,0 +1,9 @@
+name=Groovy
+extensions=groovy
+highlight_numbers=true
+singleline_string_quotes="
+singleline_comment_start=//
+multiline_comment_delims=/*, */
+multiline_string_delim="
```
Comment (ilai-deutel) [inline syntax.d/groovy.ini:7]: I'm not familiar with Groovy, but it looks like it uses """ for multi-line string literals? https://www.aurorasolutions.io/post/groovy-multi-line-strings

File: syntax.d/nix.ini [hunk 3]
```diff
@@ -0,0 +1,8 @@
+name=Nix
+extensions=nix
+highlight_numbers=true
+singleline_string_quotes="
+singleline_comment_start=#
+multiline_comment_delims=/*, */
+multiline_string_delim="
```
Comment (ilai-deutel) [inline syntax.d/nix.ini:7]: What about `''`? https://nixos.org/manual/nix/stable/language/values.html

**Review by ilai-deutel — COMMENTED** [review]
Thank you so much for your contribution!

File: syntax.d/coffeescript.ini (hunk 1 above)
Comment (TanvirOnGH) [reply syntax.d/coffeescript.ini:9]: I'm not aware of any single reference page that listed all keywords for CoffeeScript.

File: syntax.d/groovy.ini (hunk 2 above)
Comment (TanvirOnGH) [reply syntax.d/groovy.ini:7]: Ah right https://groovy-lang.org/syntax.html#_string_summary_table

**Pull request merged**